api/envoy/config/common/mutation_rules/v3/mutation_rules.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+
+package envoy.config.common.mutation_rules.v3;
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.config.common.mutation_rules.v3";
+option java_outer_classname = "MutationRulesProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// The HeaderMutationRules structure specifies what headers may be
+// manipulated by a processing filter. By default, all mutations are possible.
+// Various rules may be specified in order to restrict what a processing filter
+// can do. This structure is used by the External Processing Filter to make
+// it possible to restrict what an external processing server can do.
+message HeaderMutationRules {
+  // Prevent modification of headers that could affect routing decisions. This includes
+  // "host", ":authority", ":scheme", and ":method".
+  bool disallow_routing = 1;
+
+  // Prevent modification of envoy-internal headers. These typically start with
+  // "x-envoy" but this prefix may be overridden elsewhere.
```

Comment:
Is add and remove of headers with this prefix considered a modification or is the restriction limited to updates to the header value?

Reply:
By "modification" meant "set", "append", or remove, regardless of whether the header already exists. We COULD have separate options for all of that, but do we really need them? Otherwise I can update the docs to make it clearer.

Reply:
Clarifying the docs should be enough.

Reply:
Thanks -- I think I did that in the last commit.